### PR TITLE
netsuite custome segment import settings page changes

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

File: src/app/core/models/netsuite/netsuite-configuration/netsuite-import-setting.model.ts [hunk 1]
```diff
@@ -30,7 +32,44 @@ export type NetsuiteImportSettingConfiguration = {
...
+    updated_at?: Date;
+    workspace?: number;
+  };
+
 export class NetsuiteImportSettingModel extends ImportSettingsModel {
+
+  static getCustomeSegmentOptions(): SelectFormOption[] {
+    return [
+      {
+        label: 'Custom List',
```
Comment (ashwin1111) [inline src/app/core/models/netsuite/netsuite-configuration/netsuite-import-setting.model.ts:51]: for co, this should be sentence cased

File: src/app/core/models/netsuite/netsuite-configuration/netsuite-import-setting.model.ts [hunk 2]
```diff
@@ -30,7 +32,44 @@ export type NetsuiteImportSettingConfiguration = {
...
+      {
+        label: 'Custom Segment',
+        value: NetsuiteCustomeSegmentOption.CUSTOM_SEGMENT
+      }
+    ];
+  }
+
+  static constructCustomSegmentPayload(customeSegmentForm: FormGroup): CustomSegment {
+    return {
+      segment_type: customeSegmentForm.get('custom_field_type')?.value,
```
Comment (ashwin1111) [inline src/app/core/models/netsuite/netsuite-configuration/netsuite-import-setting.model.ts:67]: customSegmentForm

File: src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.html [hunk 3]
```diff
@@ -75,6 +75,18 @@
                         [isDestinationFixedImport]="true">
                     </app-configuration-import-field>
                 </div>
+
+                <div class="tw-mt-24-px">
+                    <div class="tw-flex tw-items-center" (click)="addCustomeSegment()">
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.html:80]: fix typo

File: src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.html [hunk 4]
```diff
@@ -75,6 +75,18 @@
...
+                        <div class="tw-mr-14-px">
+                            <app-svg-icon [svgSource]="'plus-square-medium'" [width]="'18px'" [height]="'18px'" [styleClasses]="'tw-cursor-pointer tw-text-mandatory-field-color'"></app-svg-icon>
+                        </div>
+                        <div class="tw-cursor-pointer tw-text-mandatory-field-color">
+                            Add Custom Lists/Records/Segments
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.html:85]: sentence case for c1

File: src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.html [hunk 5]
```diff
@@ -102,3 +114,7 @@
         [header]="'Preview'">
     </app-preview-dialog>
 </div>
+
+<div *ngIf="isCustomeSegmentTrigged">
+    <app-netsuite-custome-segment-dialog></app-netsuite-custome-segment-dialog>
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.html:119]: typo

File: src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.ts [hunk 6]
```diff
@@ -66,16 +67,28 @@ export class NetsuiteImportSettingsComponent implements OnInit {
 
   importSettings: any;
 
+  customrSegmentOptions: SelectFormOption[] = NetsuiteImportSettingModel.getCustomeSegmentOptions();
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.ts:70]: typo

File: src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.ts [hunk 7]
```diff
@@ -66,16 +67,28 @@ export class NetsuiteImportSettingsComponent implements OnInit {
...
+
   customFieldForm: FormGroup = this.formBuilder.group({
     attribute_type: ['', Validators.required],
     display_name: [''],
     source_placeholder: ['', Validators.required]
   });
 
+  customeSegmentForm: FormGroup = this.formBuilder.group({
+    script_id: ['', Validators.required],
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.ts:79]: use camel casing

File: src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.ts [hunk 8]
```diff
@@ -66,16 +67,28 @@ export class NetsuiteImportSettingsComponent implements OnInit {
...
+    custom_field_type: ['', Validators.required]
+  });
+
   readonly brandingFeatureConfig = brandingFeatureConfig;
 
   readonly brandingContent = brandingContent.netsuite.configuration.importSetting;
 
+  isCustomeSegmentTrigged: boolean = false;
+
+  isCustomeSegmentIsSaving: boolean = false;
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.ts:90]: isCustomSegmentSaveInProgress

File: src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.ts [hunk 9]
```diff
@@ -178,6 +199,25 @@ export class NetsuiteImportSettingsComponent implements OnInit {
...
+    this.isCustomeSegmentIsSaving = true;
+    const customeSegmentPayload = NetsuiteImportSettingModel.constructCustomSegmentPayload(this.customFieldForm);
+
+    this.importSettingService.postNetsuiteCustomSegments(customeSegmentPayload).subscribe(() => {
+      this.importSettingService.getNetsuiteFields().subscribe((netsuiteFields: IntegrationField[]) => {
+        this.isCustomeSegmentTrigged = false;
+        this.isCustomeSegmentIsSaving = false;
+        this.netsuiteFields = netsuiteFields;
+        this.toastService.displayToastMessage(ToastSeverity.SUCCESS, 'Custom field added successfully');
+
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.ts:212]: we need to setup importSettingForm again with latest field addition

File: src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.ts [hunk 10]
```diff
@@ -178,6 +199,25 @@ export class NetsuiteImportSettingsComponent implements OnInit {
...
+      });
+    }, () => {
+      this.isCustomeSegmentTrigged = false;
+      this.isCustomeSegmentIsSaving = false;
+      this.toastService.displayToastMessage(ToastSeverity.ERROR, 'Failed to add Custom Field');
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.ts:217]: sentence case for c1

File: src/app/core/models/netsuite/netsuite-configuration/netsuite-import-setting.model.ts (hunk 1 above)
Comment (DhaaraniCIT) [reply src/app/core/models/netsuite/netsuite-configuration/netsuite-import-setting.model.ts:51]: it is handled in HTML

**Comment by DhaaraniCIT** [comment]
This PR comments are fixed in [https://github.com/fylein/fyle-integrations-app/pull/776](url)

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**